### PR TITLE
Issue 178 replace exit with catch

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

replace exit and assert with throw
replace cout with LOG

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: libs/source/Socket.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ Socket::Socket(Socket::type stype) {
 
   if ((SocketFileDescriptor = socket(AF_INET, type, proto)) == -1) {
     LOG(IPC, Sev::Error, "socket() failed");
-    exit(1);
+    throw std::runtime_error("system error - socket() failed");
```
Comment (amues) [inline libs/source/Socket.cpp]: Wouldn't it be useful to log the error code or include it in the throw message in this case?

File: libs/source/Socket.cpp [hunk 2]
```diff
@@ -61,7 +61,6 @@ int Socket::setNOSIGPIPE() {
     int ret = setSockOpt(SO_NOSIGPIPE, &on, sizeof(on));
     if (ret != 0) {
         LOG(IPC, Sev::Warning, "Cannot set SO_NOSIGPIPE for socket");
-        perror("setsockopt():");
```
Comment (amues) [inline libs/source/Socket.cpp]: The same as above — isn't this information possibly helpful?

File: libs/source/Socket.cpp [hunk 3]
```diff
@@ -79,17 +78,16 @@ void Socket::setLocalSocket(const char *ipaddr, int port) {
 
   int ret = inet_aton(ipaddr, &localSockAddr.sin_addr);
   if (ret == 0) {
```
Comment (amues) [inline libs/source/Socket.cpp]: Isn't -1 also a return possibility here?

File: utils/udpredirect/udpredirect.cpp [hunk 4]
```diff
@@ -17,7 +17,7 @@ int main(int argc, char *argv[]) {
     if (argc!=3 && argc!=5) {
         printf("Usage: %s our-ip our-port send-to-ip send-to-port\n",argv[0]);
         printf("Usage: %s our-ip our-port             # echo mode\n",argv[0]);
-        exit(1);
+        return 1;
```
Comment (amues) [inline utils/udpredirect/udpredirect.cpp]: Are the only return possibilities 0 or 1? If not, using constants to codify the return codes might be a good idea.

File: libs/source/Socket.cpp (hunk 1 above)
Comment (mortenjc) [reply libs/source/Socket.cpp]: Not in this case. Either a valid socket fd is returned or -1

File: libs/source/Socket.cpp (hunk 3 above)
Comment (mortenjc) [reply libs/source/Socket.cpp]: nope: inet_aton returns 0 or 1 according to man page

File: libs/source/Socket.cpp (hunk 2 above)
Comment (mortenjc) [reply libs/source/Socket.cpp]: in this case, yes.

File: utils/udpredirect/udpredirect.cpp (hunk 4 above)
Comment (mortenjc) [reply utils/udpredirect/udpredirect.cpp]: yes, now using EXIT_SUCCESS and EXIT_FAILURE

**Pull request merged**